di/src/main/java/com/woowa/di/singleton/SingletonComponent.kt

```diff
@@ -0,0 +1,104 @@
+package com.woowa.di.singleton
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ProcessLifecycleOwner
+import com.woowa.di.ApplicationContext
+import com.woowa.di.component.Component
+import com.woowa.di.component.DIBuilder
+import com.woowa.di.findQualifierClassOrNull
+import javax.inject.Inject
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.createInstance
+import kotlin.reflect.full.declaredMemberFunctions
+import kotlin.reflect.full.hasAnnotation
+import kotlin.reflect.jvm.jvmErasure
+import kotlin.reflect.jvm.kotlinProperty
+
+class SingletonComponent<binder : Any> private constructor(private val binderClazz: KClass<binder>) :
+    Component, DefaultLifecycleObserver {
+        private lateinit var binderInstance: binder
+        private lateinit var diInstances: Map<String, Any?>
+
+        override fun onCreate(owner: LifecycleOwner) {
+            super.onCreate(owner)
+            binderInstance = binderClazz.createInstance()
+            diInstances = saveDIInstance()
+        }
+
+        override fun onDestroy(owner: LifecycleOwner) {
+            super.onDestroy(owner)
+            instances.remove(binderClazz)
+            owner.lifecycle.removeObserver(this)
+        }
```

Comment:
저도 고민중인 사항인데, 벼리의 생각이 궁금하여 코멘트 한번 남깁니다! 😂
```suggestion
        override fun onDestroy(owner: LifecycleOwner) {
            instances.remove(binderClazz)
            owner.lifecycle.removeObserver(this)
            super.onDestroy(owner)
        }
```
혹시 두 코드의 동작 상 차이점이 어떤 것이 있는지 알고 계신 것이 있는지 궁금합니다! 
해당 순서로 구현하신 특별한 기준이 있으신지 궁금해서요!

지난 수업에서 ViewModel 클래스를 직접 만들 때는 onDestroy() 콜백에서 부모 클래스 관련 처리(super.onDestroy)가 필요한 작업 이후 이뤄지고 있었는데(다른 해제 작업 이후 onDestroy 호출), 
이러한 순서의 차이가 컴포넌트가 어떤 것인지 그리고 어떤 작업을 하려고 하는지에 따라 갈릴 수 있다는 생각이 들었던 것 같아요.
여유가 되시면 한 번 답변을 부탁드리겠습니다😆